### DISCUSSION TITLE
clusterProfiles utilities

### DISCUSSION THREAD

**Discussion by mariabernard** [YuLab-SMU/clusterProfiler#506] [Q&A] [answered]
Hi,

I am working on metagenomic data. I have predicted KEGG functions associated to bacteria, and I have done a differential analysis on these functions using DESeq2.

I used clusterProfiler to get enrichmentAnalysis on differential abundat function, and I got intesresting results (many thanks for your tool!!) using this type of command:

```
all_enrich_func <- enrichKEGG(
  all_da_kegg_fun,
  organism = "ko",
  universe = all_kegg_func,
  minGSSize = 10,
  maxGSSize = 500,
  pvalueCutoff = 1,
  qvalueCutoff = 1,
  pAdjustMethod = "BY",
  use_internal_data = FALSE
)
```

Now I am looking for a tool (or parsing code), to associate each (and all ) Kegg number to patway KO and MAP number (Id reported in Ipath3 visualisation interface) and description.

Something like:
```
K_number KO_number	MAP_number	Description
K01034	 ko00650	map00650	Butanoate metabolism
K01034	 ko00310	map00310	Lysine degradation
K01034	 ko01120	map01120	Microbial metabolism in diverse environments
K01034	 ko02020	map02020	Two-component system
```
In my example ko are equal to map. I don't know if it's always the case ?!

As the result of enrichKEGG returns almost these informations, I guess that clusterProfiler is able to query KEGG databse to get these information, but I did not find this kind of tool in the clusterProfiler utility.
Or maybe you have a parsing code to convert 
```
ID      Description          geneID
ko00650 Butanoate metabolism K01039/K01040/K00248/K01034/K01715/K01035/K00074/K14534/K00626/K01615/K00929/K01640/K00174/K00175/K18120/K01029/K00132
```
in the expected format ?

sorry I'm thinking about it as I write ...

Many thanks for your idea.

Maria

**Comment by huerqiang** [comment, marked as answer]
This should be what you need:
```r
kegg <- download_KEGG("ko", "KEGG", "kegg")
```

**Reply by mariabernard** [reply]
Thanks, I should be fine with that.
